File: GXpert/GXpert.Web/Modules/ServerTypes/QuestionBank/CommonDataColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CommonDataRow } from "./CommonDataRow";

export interface CommonDataColumns {
    Id: Column<CommonDataRow>;
    CommonDataTitle: Column<CommonDataRow>;
    CommonData: Column<CommonDataRow>;
}

export class CommonDataColumns extends ColumnsBase<CommonDataRow> {
    static readonly columnsKey = 'QuestionBank.CommonData';
    static readonly Fields = fieldsProxy<CommonDataColumns>();
}

File: GXpert/GXpert.Web/Modules/QuestionBank/CommonData/CommonDataColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.QuestionBank.Columns;

[ColumnsScript("QuestionBank.CommonData")]
[BasedOnRow(typeof(CommonDataRow), CheckNames = true)]
public class CommonDataColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string CommonDataTitle { get; set; }
    [EditLink]
    public string CommonData { get; set; }
}